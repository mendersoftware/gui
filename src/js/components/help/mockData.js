// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
