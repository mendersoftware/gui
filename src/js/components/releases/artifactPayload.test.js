import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import ArtifactPayload from './artifactPayload';

describe('ArtifactPayload Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<ArtifactPayload payload={{ files: null, type_info: { type: 'test' } }} />);
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
