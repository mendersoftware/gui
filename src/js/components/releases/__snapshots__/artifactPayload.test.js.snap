// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtifactPayload Component renders correctly 1`] = `
.emotion-0 {
  background: #e9e9e9;
  margin: 0;
  padding: 10px;
  position: absolute;
  top: -20px;
}

.emotion-1 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 11px;
  padding-bottom: 11px;
  padding-bottom: 11px;
  border-bottom: 1px solid #cfcfcf;
  margin-right: 2vw;
}

.emotion-2.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-2.Mui-selected {
  background-color: rgba(51, 122, 135, 0.08);
}

.emotion-2.Mui-selected.Mui-focusVisible {
  background-color: rgba(51, 122, 135, 0.2);
}

.emotion-2.Mui-disabled {
  opacity: 0.38;
}

.emotion-2.Mui-disabled {
  opacity: 1;
}

.emotion-2.active {
  background-color: #fff;
}

.emotion-2.leftNav.active {
  border-top: 1px solid #d4e9e7;
  border-bottom: 1px solid #d4e9e7;
}

.emotion-2.navLink,
.emotion-2.navLink .MuiListItemText-root {
  color: #969696;
}

.emotion-3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 6px;
  margin-bottom: 6px;
  margin-top: 0;
  margin-bottom: 0;
  color: rgba(10, 10, 11, 0.78);
}

.emotion-4 {
  margin: 0;
  line-height: 1.5;
  font-family: Lato,sans-serif;
  font-weight: 400;
  font-size: 0.9286rem;
  display: block;
}

.emotion-5 {
  margin: 0;
  font-family: Lato,sans-serif;
  font-weight: 400;
  font-size: 0.8125rem;
  line-height: 1.43;
  color: rgba(0, 0, 0, 0.54);
  display: block;
}

<div
  class="file-details"
>
  <h4
    class="emotion-0"
  >
    Payload 
  </h4>
  <ul
    class="MuiList-root MuiList-padding emotion-1"
  >
    <li
      class="MuiListItem-root attributes MuiListItem-gutters MuiListItem-padding Mui-disabled opaque emotion-2"
      disabled=""
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline emotion-3"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
        >
          Type
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-5"
        >
          test
        </p>
      </div>
    </li>
    <li
      class="MuiListItem-root attributes MuiListItem-gutters MuiListItem-padding Mui-disabled opaque emotion-2"
      disabled=""
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline emotion-3"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
        >
          Total Size (uncompressed)
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-5"
        >
          0 Bytes
        </p>
      </div>
    </li>
  </ul>
  <div
    class="file-meta"
  >
    <h4>
      Files
    </h4>
    <p>
      There are no files in this Artifact
    </p>
  </div>
</div>
`;
